### DISCUSSION TITLE
How can i solve conflicting values error

### DISCUSSION THREAD

**Discussion by xi3161-sregoti** [cue-lang/cue#1755] [Q&A] [answered]
The below code
```
a: {
    "foo": 1
    ...
}

b: a & {
    "foo": 2
}
```

Gives this error
```
b.foo: conflicting values 2 and 1:
    ./data.cue:9:12
    ./data.cue:13:4
    ./data.cue:14:12
 ```
 
How can i resolve this error, I want to override the default value present in the object `a`

**Comment by slewiskelly** [comment, marked as answer]
Overriding concrete values in CUE is not possible, see [the documentation](https://cuelang.org/docs/concepts/logic/#relation-to-inheritance).

You can achieve what you want by using default values, instead:

```cue
a: {
    "foo": int | *1
    ...
}

b: a & {
    "foo": 2
}
```

**Reply by xi3161-sregoti** [reply]
Even this works for me. Thanks

**Comment by georgfaust** [comment]
No overrides is one of the core features of cue.
I strongly recommend to go through the documentation, you wont have any fun otherwise. Cue is different.
